osu.Game/Screens/SelectV2/BeatmapLeaderboardWedge.cs

```diff
@@ -0,0 +1,374 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Cursor;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Localisation;
+using osu.Game.Online.Leaderboards;
+using osu.Game.Online.Placeholders;
+using osu.Game.Overlays;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Scoring;
+using osu.Game.Screens.Select.Leaderboards;
+
+namespace osu.Game.Screens.SelectV2
+{
+    public partial class BeatmapLeaderboardWedge : VisibilityContainer
+    {
+        private Container scoresContainer = null!;
+
+        private OsuScrollContainer scoresScroll = null!;
+        private Container personalBestDisplay = null!;
+        private Container<BeatmapLeaderboardScore> personalBestScoreContainer = null!;
+        private LoadingLayer loading = null!;
+
+        private Container<Placeholder> placeholderContainer = null!;
+        private Placeholder? placeholder;
+
+        [Resolved]
+        private LeaderboardManager leaderboardManager { get; set; } = null!;
+
+        [Resolved]
+        private IBindable<WorkingBeatmap> beatmap { get; set; } = null!;
+
+        [Resolved]
+        private IBindable<RulesetInfo> ruleset { get; set; } = null!;
+
+        [Resolved]
+        private IBindable<IReadOnlyList<Mod>> mods { get; set; } = null!;
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
+        public IBindable<BeatmapLeaderboardScope> Scope { get; } = new Bindable<BeatmapLeaderboardScope>();
+
+        public IBindable<bool> FilterBySelectedMods { get; } = new BindableBool();
+
+        private CancellationTokenSource? cancellationTokenSource;
+
+        private readonly IBindable<LeaderboardScores?> fetchedScores = new Bindable<LeaderboardScores?>();
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            RelativeSizeAxes = Axes.Both;
+
+            Child = new OsuContextMenuContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Children = new Drawable[]
+                {
+                    scoresScroll = new OsuScrollContainer
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        ScrollbarVisible = false,
+                        Shear = OsuGame.SHEAR,
+                        Child = scoresContainer = new Container
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Margin = new MarginPadding { Top = 4f, Bottom = 180f },
```

Comment:
Don't want to get too involved in the design side of things, but what is this bottom margin spec doing? I couldn't help but notice when checking the test scene that the margin makes it so that when you scroll all the way down there's a huge gap:

![Screenshot 2025-04-23 at 13 23 06](https://github.com/user-attachments/assets/146af139-e650-4c6f-addb-75df7090ed3d)

Seems much too weird for me to not say anything?

Reply:
It's enough bottom bleed to allow the last card to show fully without being cut off from the left side.

Reply:
Should probably be commented as such!

Reply:
Has been resolved in https://github.com/ppy/osu/pull/32844/commits/bd58aac9cca8b410b550ce6fa8afcdfb68a14b5d